Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesInterface.h

```diff
@@ -55,9 +61,15 @@ namespace AzFramework
 
         const AZ::Entity* const* begin();
         const AZ::Entity* const* end();
+        const AZ::Entity* const* begin() const;
+        const AZ::Entity* const* end() const;
         const AZ::Entity* const* cbegin();
         const AZ::Entity* const* cend();
-        size_t size();
+
+        const AZ::Entity* operator[](size_t n);
+        const AZ::Entity* operator[](size_t n) const;
+
+        size_t size() const;
```

Comment:
You might want to add an `[[nodiscard]] bool empty() const` function as well.
That provides interoperability with [std::empty](https://en.cppreference.com/w/cpp/iterator/empty)

Reply:
Done